test/unittest/memory/unittest_cache_pool.cpp

```diff
@@ -0,0 +1,328 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2022 Jiho Chu <jiho.chu@samsung.com>
+ *
+ * @file unittest_cache_pool.cpp
+ * @date 28 Sep 2022
+ * @brief Cache Pool Test
+ * @see	https://github.com/nnstreamer/nntrainer
+ * @author Jiho Chu <jiho.chu@samsung.com>
+ * @bug No known bugs except for NYI items
+ */
+
+#include "optimized_v1_planner.h"
+#include <cstring>
+#include <random>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <basic_planner.h>
+#include <cache_pool.h>
+#include <nntrainer_test_util.h>
+
+constexpr float TEMP_DATA1 = (12345.12345);
+constexpr float TEMP_DATA2 = (67890.67890);
+constexpr float TEMP_DATA3 = (54321.54321);
+
+/**
+ * @brief Cache pool mock class
+ */
+class MockCachePool : public nntrainer::CachePool {
+public:
+  MockCachePool(const std::string &name) : nntrainer::CachePool(name) {
+    ON_CALL(*this, validate).WillByDefault([&](unsigned int id) {
+      nntrainer::CachePool::validate(id);
+    });
+    ON_CALL(*this, invalidate).WillByDefault([&](unsigned int id) {
+      nntrainer::CachePool::invalidate(id);
+    });
+  }
+
+  /**
+   * @brief validate mock method
+   */
+  MOCK_METHOD(void, validate, (unsigned int id), (override));
+
+  /**
+   * @brief invalidate mock method
+   */
+  MOCK_METHOD(void, invalidate, (unsigned int id), (override));
+};
+
+/**
+ * @brief Cache pool test class
+ */
+class CachePoolTest : public ::testing::Test {
+public:
+  void SetUp(void) { pool = new MockCachePool("tmp pool"); }
+
+  void TearDown(void) {
+    EXPECT_NO_THROW(pool->clear());
+
+    delete pool;
+  }
+
+  MockCachePool *pool;
+};
+
+/**
+ * @brief creation and destruction
+ */
+TEST_F(CachePoolTest, create_destroy) {}
+
+/**
+ * @brief get cache memory
+ */
+TEST_F(CachePoolTest, get_memory_01_p) {
+  EXPECT_CALL(*pool, validate).Times(0);
+  EXPECT_CALL(*pool, invalidate).Times(testing::AtLeast(1));
+
+  std::shared_ptr<nntrainer::MemoryData<float>> mem;
+  auto idx = pool->requestMemory(1, 4, 5);
+  EXPECT_NO_THROW(pool->planLayout(nntrainer::BasicPlanner()));
+  EXPECT_NO_THROW(pool->allocate());
+
+  EXPECT_NO_THROW(mem = pool->getMemory(idx));
+  EXPECT_NE(mem, nullptr);
+  EXPECT_EQ(mem->getAddr(), nullptr);
```

Comment:
I think it could be great if more details are provided.

Reply:
thanks. comments which describes each test case are added.